Parser/Parser.cs

```diff
@@ -67,12 +70,28 @@ NExpr Expression ()
       return expr;
    }
 
+   NExpr[] ArgList () {
+      List<NExpr> args = new ();
+      for (; ;) {
+         if (Match (EOF)) throw new Exception ($"Expecting ')'");

```

Comment:
There is no other place in the parser where we have to check for EOF. Can you rewrite so this Match(EOF) is not required?

Reply:
@tarydon: Rewrote the above method.